grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -75,9 +77,14 @@
   @GuardedBy("bindLock")
   final Map<String, ChannelRef> affinityKeyToChannelRef = new HashMap<String, ChannelRef>();
 
+  // Map from a broken channel id to the remapped affinity keys (key => ready channel id).
+  @GuardedBy("this")
+  private final HashMap<Integer, HashMap<String, Integer>> fallbackMap = new HashMap<>();
```

Comment:
ConcurrentHashMap seems faster in this code for this object.

(I don't know why the concept of declaring of a bindLock was used in this code, instead simply using a Synchronized HashMap (or better yet, use ConcurrentHashMap if it can serve the purpose - because ConcurrentHashMap is faster)

Reply:
Oh, right. ConcurrentHashMap is a perfect match for my use-case. Thanks, I will remake that!
I guess we can use ConcurrentHashMap for the `affinityKeyToChannelRef` too, but I will research that later.

Reply:
Replaced with ConcurrentHashMap.